src/tribler-common/tribler_common/patch_import.py

```diff
@@ -41,11 +41,11 @@ def try_import(module_name, *args, **kwargs):
             raise ImportError
 
         if strict:
-            return mock.MagicMock(**mock_kwargs)
+            return Mock(**mock_kwargs)
```

Comment:
**Question**: What is the reason to change `MagicMock` to `Mock`?

Reply:
https://stackoverflow.com/questions/17181687/mock-vs-magicmock

In short, regular `Mock` does **not** provide so-called "magic methods". This has two advantages:
1. it is a bit lighter
2. if you don't _need_ to mock magic methods, you really **should not** mock them (e.g. you don't want your objects to behave like a container, don't you? :wink: )

Reply:
> it is a bit lighter

True

>if you don't need to mock magic methods, you really should not mock them (e.g. you don't want your objects to behave like a container, don't you? 😉 )

In the method above `MagicMock` is used as a mock class for the library patch. Therefore `MagicMock` will be returned as a library replacement.

In the case of the following library usage, `MagicMock` gives you a correct behavior:

```python
    import library
    print(len(library.some_set))
```

With your improvement, the example above will fail.

Reply:
> With your improvement, the example above will fail.

And that is exactly what we want. Tests must fail when something unexpected happens. When that happens, we replace the offending mock with `MagicMock`. But at least we do that for a reason.

Reply:
I'm not sure about your intention, but my intention was the following. I introduced this mechanism mostly for mocking libraries during swagger extraction (doc build).

You can not guarantee that libraries from our `requerements.txt` are not used in the way that I described above.

Also, it should never fail for the example above even for the tests because it is not unexpected behavior. It is expected behavior.

Reply:
>Also, it should never fail for the example above even for the tests because it is not unexpected behavior. It is expected behavior.


Good point, I overlooked that.

Reply:
I will add a test to document desirable behavior after your PR be merged.

Reply:
Here is the tests improvement: https://github.com/Tribler/tribler/pull/6588